### DISCUSSION TITLE
can not get my button to add a record to the Table

### DISCUSSION THREAD

**Discussion by HFolkertH** [cincheo/daquota#14] [Q&A] [answered]
hello Renaud,

I tried to learn from  the examples but Dlite still has some mysteries.
I can not get my add-button to add a book to this simple books-table. What am i missing?
I added a copy of the books.dlite file. By the way: i need to change the file name by adding a.txt extension. Otherwise the dlite file can not be added to this discussion.

kind regards
Folkert
[books.dlite.txt](https://github.com/cincheo/dlite/files/8728770/books.dlite.txt)

**Comment by renaudpawlak** [comment, marked as answer]
[books-fixed.dlite.txt](https://github.com/cincheo/dlite/files/8736124/books-fixed.dlite.txt)

Here is the fixed version.
There were several issues, but the most important (which is a common mistake), is that the data in your local storage was wrong. It was initialized with a string "[]", instead of an actual array. To initialize with an empty array, you need to make sure, you press the formula button when you enter the "Default value", so that it is initialized with a JavaScript object (otherwise it is always a string). 

Now, since you have initialized your data with a string instead of an array, your button @click event was not able to use the "addData" action. Instead, you used the "setData". So I changed it to "addData".

You will have to reset the value in your local storage to start with new data (because you will probably have an object in it). To do so, in "local-storage-0", just press the "Reset data" button, and it should take the default value again (an empty array).

From there, everything should work as expected.

To avoid that kind of confusion in the future, I will change the default initialization of a storage to have an empty array by default, because I believe that's what people need by default like 90% of the time.

**Reply by HFolkertH** [reply]
hi Renaud,

Thanks. I already figured out that I needed an empty array,  but i did not know it should be done with the formula editor.
